src/github.com/matrix-org/dendrite/mediaapi/writers/utils.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writers
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/dendrite/mediaapi/types"
+	"github.com/matrix-org/util"
+)
+
+// createTempDir creates a tmp/<random string> directory within baseDirectory and returns its path
+func createTempDir(baseDirectory types.Path) (types.Path, error) {
+	baseTmpDir := path.Join(string(baseDirectory), "tmp")
+	err := os.MkdirAll(baseTmpDir, 0770)
+	if err != nil {
```

Comment:
Style: For single-value error returns with a `nil` check we nearly always in-line it:
```go
if err := os.MkdirAll(baseTmpDir, 0770); err != nil {
    // ...
}
```

The only times we don't is for multi-value returns (scoping is wrong) and for cases where the function call is really long. The main reason why we opt to do this is to try to represent one 'action' succinctly, and rarely for variable scoping reasons.